crates/core/src/transcript/encoding/proof.rs

```diff
@@ -182,3 +182,92 @@ impl From<MerkleError> for EncodingProofError {
         Self::new(ErrorKind::Proof, error)
     }
 }
+
+/// Invalid encoding proof error.
+#[derive(Debug, thiserror::Error)]
+#[error("invalid encoding proof: {0}")]
+pub struct InvalidEncodingProof(&'static str);
+
+mod validation {
+    use super::*;
+
+    /// The maximum allowed height of the Merkle tree of encoding commitments.
+    ///
+    /// The statistical security parameter (SSP) of the encoding commitment
```

Comment:
just curious on this statement about how SSP is calculated for the encoding commitment protocol: is there a paper/textbook that outlines this or its our team's finding?

Reply:
This is not documented. Maybe we should document it. But this commitment protocol is kinda still not solidifed to document it atm.

The SSP of our encoding commitment protocol is essentially "what is the probabily that a prover who tries to guess the encoding of a single bit will succeed". The success probability is 1/2^SSP.

Suppose, our encodings are 40 bits, then the probability of the prover guessing an encoding of a sinlge bit is 1/2^40. That gives us protocol 40 bits of statistical security.

Suppose, a malicious prover fills the Merkle tree with 2^10 1-bit encodings which he guessed at random. Later, when the prover learns what the encoding seed is, he will check if any of his 2^10 guesses matches the encoding and. If it matches, then the attack succeeded and the prover will reveal that specific Merkle tree leaf which matched to the verifier.

The fact the the prover was allowed to make 2^10 guesses brought down the SSP from 40 to 30. Hence, we need to put a cap on the size of the Merkle tree and also require that the bit-length of a single encoding is not too small. We need to achieve at least SSP = 40.

Reply:
tthanks for the explanation!